src/ucp/proto/proto_common.c

```diff
@@ -806,7 +807,8 @@ void ucp_proto_request_restart(ucp_request_t *req)
     }
 
     /* Select a protocol with resume request support */
-    if (!ucp_datatype_iter_is_begin(&req->send.state.dt_iter)) {
+    if (ucp_proto_multi_is_middle_fragment(req) ||
```

Comment:
this is active-message specific, so maybe the protocol "reset" callback should provide the indication of whether a "RESUME" flags is going to be required or not, instead of just deciding it according to ucp_datatype_iter_is_begin

Reply:
I reduced the scope of the changes to AM eager multi, and I don't see test failures in this function